### PR TITLE
More control over creation/removal of DMS components

### PR DESCRIPTION
**Pull request by nikokaoja**
## Added
- Ability to upload of all spaces components or only ones that are in space defined by `Rules.metadata.space`
- Ability to remove of all spaces components or only ones that are in space defined by `Rules.metadata.space`

## Improved
- DMS Schema components upload report add to step `ExportDMSSchemaComponentsToCDF`
- DMS Schema components removal report add to step `DeleteDMSSchemaComponents`
- Handling of multiple steps

## Removed
- `DataModelFromRulesToSourceGraph` it is confusing step and needs more work to be useful
- Workflows:
  - `json_to_data_model_rules`
  - `sheet2cdf`
  - `skos2cdf`

## Changed
- Renamed steps:
  - `LoadTransformationRules` to `ImportExcelToRules`
  - `InstancesFromRdfFileToSourceGraph` to `ExtractGraphFromRdfFile`
  - `InstancesFromRulesToSolutionGraph` to `ExtractGraphFromRulesInstanceSheet`
  - `GraphCapturingSheetToGraph` to `ExtractGraphFromGraphCapturingSheet`
  - `GenerateMockGraph` to `ExtractGraphFromMockGraph`
  - `InstancesFromJsonToGraph` to `ExtractGraphFromJsonFile`
  - `InstancesFromAvevaPiAF` to `ExtractGraphFromAvevaPiAssetFramework`
  - `DexpiToGraph` to `ExtractGraphFromDexpiFile`
  - `GenerateCDFAssetsFromGraph` to `GenerateAssetsFromGraph`
  - `GenerateCDFRelationshipsFromGraph` to `GenerateRelationshipsFromGraph`
  - `GenerateCDFNodesAndEdgesFromGraph` to `GenerateNodesAndEdgesFromGraph`
  - `UploadCDFAssets` to `LoadAssetsToCDF`
  - `UploadCDFRelationships` to `LoadRelationshipsToCDF`
  - `UploadCDFNodes` to `LoadNodesToCDF`
  - `UploadCDFEdges` to `LoadEdgesToCDF`
  - `CreateCDFLabels` to `LoadLabelsToCDF`
  - `OpenApiToRules` to `ImportOpenApiToRules
  - `ArbitraryJsonYamlToRules` to `ImportArbitraryJsonYamlToRules`
  - `GraphToRules` to `ImportGraphToRules`
  - `OntologyToRules` to `ImportOntologyToRules`
  - `GraphQLSchemaFromRules` to `ExportGraphQLSchemaFromRules`
  - `OntologyFromRules` to `ExportOntologyFromRules`
  - `SHACLFromRules` to `ExportSHACLFromRules`
  - `GraphCaptureSpreadsheetFromRules` to `ExportRulesToGraphCapturingSheet`
  - `ExcelFromRules` to `ExportRulesToExcel`
- Renamed workflows:
  - `graph_to_asset_hierarchy` to `extract_rdf_graph_generate_assets`
  - `dexpi2graph` to `extract_dexpi_graph_and_export_rules`
  - `ontology2data_model` to `import_ontology`

- **Note** this is a breaking change, but since we are on 0. version, we can do this.

### CODE REVIEW

File: cognite/neat/rules/exporter/_rules2dms.py [hunk 1]
```diff
@@ -546,171 +566,312 @@ def to_cdf(
...
+        existing_containers: set,
+        update: bool = False,
+        multi_space_components_create: bool = False,
+    ) -> tuple[list, list]:
+        logs, errors = [], []
+
+        containers_to_create = (
+            set(self.containers.keys()) - existing_containers
+            if multi_space_components_create
+            else {k for k in (set(self.containers.keys()) - existing_containers) if k.split(":")[0] == self.space}
```
Comment (doctrino) [inline cognite/neat/rules/exporter/_rules2dms.py:647]: Note sure if this is the PR, but code like this is fairly risky `k.split(":")[0] == self.space`. Suggest that we use `ContainerId` instead

**Review by doctrino — APPROVED** [review]

File: cognite/neat/workflows/steps/lib/rules_importer.py [hunk 2]
```diff
@@ -3,22 +3,124 @@
...
+                store.load_rules_file_from_cdf(str(rules_file), version)
+        else:
+            store.load_rules_file_from_cdf(str(rules_file), version)
+
+        raw_rules = importer.ExcelImporter(rules_file_path).to_raw_rules()
+        rules, errors, warnings_ = raw_rules.to_rules(return_report=True, skip_validation=False)
+        report = "# RULES VALIDATION REPORT\n\n" + generate_exception_report(errors, "Errors")
+
+        with report_full_path.open(mode="w") as file:
+            file.write(report)
```
[the remark this answers is not in the document]
Comment (nikokaoja) [reply cognite/neat/workflows/steps/lib/rules_importer.py:95]: We used it already in the code based :) , the idiot (me) copy/paste the above code from another step :)

File: cognite/neat/workflows/steps/lib/rules_importer.py [hunk 3]
```diff
@@ -3,22 +3,124 @@
...
 from cognite.neat.workflows.steps.data_contracts import RulesData, SolutionGraph, SourceGraph
 from cognite.neat.workflows.steps.step_model import Configurable, Step
 
 CATEGORY = __name__.split(".")[-1].replace("_", " ").title()
 
-__all__ = ["OpenApiToRules", "ArbitraryJsonYamlToRules", "GraphToRules", "OntologyToRules"]
+__all__ = ["ExcelToRules", "OpenApiToRules", "ArbitraryJsonYamlToRules", "GraphToRules", "OntologyToRules"]
+
+
+class ExcelToRules(Step):
```
[the remark this answers is not in the document]
Comment (nikokaoja) [reply cognite/neat/workflows/steps/lib/rules_importer.py:29]: Good point, actually I've renamed all the steps in that module prefixing their names with Import

File: cognite/neat/rules/exporter/_rules2dms.py [hunk 4]
```diff
@@ -546,171 +566,312 @@ def to_cdf(
...
+        update: bool = False,
+        multi_space_components_create: bool = False,
+    ) -> tuple[list, list]:
+        logs, errors = [], []
+
+        spaces_to_create = (
+            self.spaces - existing_spaces
+            if multi_space_components_create
+            else ((self.spaces - existing_spaces) & {self.space})
+        )
```
[the remark this answers is not in the document]
Comment (nikokaoja) [reply cognite/neat/rules/exporter/_rules2dms.py:615]: Good one !

File: cognite/neat/rules/exporter/_rules2dms.py (hunk 1 above)
Comment (nikokaoja) [reply cognite/neat/rules/exporter/_rules2dms.py:647]: Let's hit this one in another PR, I was not extremely happy with this either, but will solves issues now/unblocks clients

**Pull request merged**